### DISCUSSION TITLE
Trouble getting anything besides text when using V2 and get_users_tweets() method

### DISCUSSION THREAD

**Discussion by andycccccccc** [tweepy/tweepy#1756] [Questions] [answered]
Hello,

Currently trying to migrate over to API v2, but I'm having difficulty getting the usual extras when querying user timelines. API v1.1 automatically returns fields such as "created_at", and retweet/reply/quote statuses, but apparently V2 needs to be told explicitly to return those items. The issue I'm running into is that the procedures for returning that data as outlined in official documentation do not seem to be working for Tweepy.

Here is my code, attempting to get both "created_at" and "public_metrics" in addition to the tweet text:

```
import Tweepy
TOKEN = '12345678912345678912345567889etc'
id = '1234567890'

tweepy_client = authorize_twitter_oauth2(TOKEN) #just a custom login function
tweets = tweepy_client.get_users_tweets(id=id, tweet_fields=['created_at,public_metrics']).data
```

The above code just returns the tweet id and the tweet text, ignoring my requests for "created_at" and "public_metrics". I've also tried formatting the "tweet_fields" argument like this, but still to no avail:
```
tweet_fields=['created_at' ,'public_metrics']
```

Am I doing something wrong or is this a current limitation of Tweepy?

Thanks for any support.

**Comment by Harmon758** [comment, marked as answer]
https://tweepy.readthedocs.io/en/v4.5.0/faq.html#why-am-i-not-getting-expansions-or-fields-data-with-api-v2-using-client

**Reply by andycccccccc** [reply]
Hi, thanks for the reply, but I'm still having issues. According to the link, supplying the .data attribute to the returned timeline object should show the additional fields, but it's still only returning id and text. This is how I've been putting it:
```
tweets = tweepy_client.get_users_tweets(id=id, tweet_fields=['created_at,public_metrics'])
tweet_data = tweets.data
```

 I've also tried supplying other attributes that might fit the bill, like:
```
tweets = tweepy_client.get_users_tweets(id=id, tweet_fields=['created_at,public_metrics']).created_at
```
 but this throws errors. Wish there were some examples in the docs for this v2 endpoint.

Does anyone have a working example of successfully getting the 'created_at' field? That would help me out huge. 

Thanks.

**Reply by Harmon758** [reply]
The returned object, `tweets`, is a [`Response`](https://tweepy.readthedocs.io/en/latest/client.html#response) object. It is not a Tweet object and therefore will not have Tweet object attributes.
You're accessing the `data` attribute of that `Response` object, which is a list of [Tweet objects](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet).
As the FAQ section says, the string representation of each Tweet object only includes the default `id` and `text` fields, so if you're only printing the list of Tweet objects, that's all you'll see for each Tweet.

**Reply by andycccccccc** [reply]
> The returned object, `tweets`, is a [`Response`](https://tweepy.readthedocs.io/en/latest/client.html#response) object. It is not a Tweet object and therefore will not have Tweet object attributes. You're accessing the data attribute of that `Response` object, which is a list of [Tweet objects](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet). As the FAQ section says, the string representation of each Tweet object only includes the default `id` and `text` fields, so if you're only printing the list of Tweet objects, that's all you'll see for each Tweet.

Thanks, is the way to do it then to access the first tweet in the response object as if it was a list? Trying this also does not work for me, unfortunately, throws an error:
```
tweets = tweepy_client.get_users_tweets(id=id, tweet_fields=['created_at,public_metrics'])
tweet_data = tweets[0].data
```
At this point I'm unsure how to access the tweet object. Your guidance is great, but feels like I'm really just guessing at the proper implementation of this stuff. It's very different from 1.1 and there aren't any examples in the docs.

**Reply by Harmon758** [reply]
The [`Response`](https://tweepy.readthedocs.io/en/latest/client.html#response) object is a named tuple, not a list. `Response.data` is the data returned by the API request. In this case, that's a list of Tweet objects. You can access and use that list, e.g.:
```py
response = tweepy_client.get_users_tweets(id=id, tweet_fields=['created_at', 'public_metrics'])
tweets = response.data
tweet = tweets[0]
tweet_data = tweet.data
```

**Reply by andycccccccc** [reply]
> 

That worked perfectly. Thank you very much for the working example, and for taking the the time with me. Really helped me understand the conventions better. Much appreciated!

**Reply by global-live** [reply]
Thank you as well :) You`ve been very helpful for some newbs like me.
